### PR TITLE
[form-text-area] Update inline min-height on component update

### LINKED ISSUE #2496: Textarea - min-height inline style not adjusted with rows
# Bug Report

## Description
  During 'Textarea' creation, a min-height is assigned dependent on the rows or size prop.  When a Textarea is moved or swapped, it doesn't require a full re-render and so only prop changes will be updated by React.  This leaves inline styling like min-height at values that disrupt the UI.  Namely, a smaller sized Textarea will become the height of a larger sized Textarea if they swap so that it meets the min-height value.  

## Steps to Reproduce

1.  Create two components containing 'TextAreas' with a size difference, either through prop.size or prop.rows.  props- 'value', 'classname', have the same values, 'isAutoResizable' and 'disabled' are set to true.
2.  Display the components, the Textareas should be correctly sized
3.  Swap the components, now the Textareas should have equal sized Textareas

## Additional Context / Screenshots

Repo: [https://github.training.cerner.com/DevCenter/CernerSurveySystem/tree/stable](url)
Branch using Textarea: https://github.training.cerner.com/DevCenter/CernerSurveySystem/blob/stable/src/Cerner.SurveySystem.Service/wwwroot/javascript/components/Questions/Question.jsx#L275
The swapping occurs in SurveyForm.jsx

textAnswersSize

Related Jira - https://jira2.cerner.com/browse/ACADEM-46920

## Expected Behavior

Changing Textarea sizes should keep the correct heights for both Textareas.

## Possible Solution

Textarea's SetBaseHeight() and/or resizeTextarea() could update the inline style?

## Environment

* Component Name and Version: terra-core-textarea
* Browser Name and Version: Chrome 75.0.3770.100 (official) 64-bit
* Node/npm Version: Node 10.16.0/npm 6.9.1
* Webpack Version: -
* Operating System and version (desktop or mobile): MS10 Enterprise 10.0.17763

## @ Mentions

@tbiethman

### PR DESCRIPTION
**Pull request by yuderekyu**
### Summary
Closes #2496. Adds logic to update min-height when component updates.

### CODE REVIEW

File: packages/terra-form-textarea/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ Unreleased
 ### Added
 * Added documentation on required peerDependencies
 
+### Fixed
+* Fix min-height inline style bug - Use componentDidUpdate update to apply min height
```
Comment (StephenEsser) [inline packages/terra-form-textarea/CHANGELOG.md]: This snuck into a released section

Comment (yuderekyu) [reply packages/terra-form-textarea/CHANGELOG.md]: moved to unreleased

File: packages/terra-form-textarea/src/Textarea.jsx [hunk 2]
```diff
@@ -114,6 +114,17 @@ class Textarea extends React.Component {
...
   }
 
+  // Addresses https://github.com/cerner/terra-core/issues/2496.
+  // min-height manipulation exists outside the virtual dom. React reconcilliation will fail to update this inline style, because it is not a prop.
+  // This is apparent when swapping different sized TextArea components. To alleviate this, first reset the min-height.
+  // This allows us to capture the correct scrollHeight, and use this correct value for min-height.
+  componentDidUpdate(prevProps) {
+    if (this.props.size !== prevProps.size || this.props.rows !== prevProps.rows) {
+      this.textarea.style.minHeight = '0px';
+      this.textarea.style.minHeight = `${this.textarea.scrollHeight}px`;
```
[the remark this answers is not in the document]
Comment (yuderekyu) [reply packages/terra-form-textarea/src/Textarea.jsx]: I'll move the reset logic into `setBaseHeight`, then call `setBaseHeight` here like you suggested

**Comment by yuderekyu** [comment]
@ryanthemanuel I added documentation to address resizing textareas with the gripper.

https://github.com/cerner/terra-core/pull/2528/commits/85b8f94298e529481fd5a0f0759abe34362301ba

**Pull request merged**